File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit,cost;
        int buy = prices[0];
        profit = 0;
        for(int i=1;i<prices.length;i++)
        {
            cost = prices[i]-buy;
            profit = Math.max(profit,cost);
            buy = Math.min(buy,prices[i]);
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    min_p := prices[0]
    res := 0
    for i:=1 ; i<len(prices);i++ {
        if min_p > prices[i] {
            min_p = prices[i];
        }
        profit :=  prices[i] - min_p
        if res < profit {
            res = profit
        } 
    }
    return res
}